bin/pynumenc

```diff
@@ -2,75 +2,108 @@
+            "from_type, got {}".format(conversion))
+        return 1
 
-    if tajp not in supported_types:
-        raise ValueError("Type {} is not supported. The supported types are:\n "
-                         "{}.".format(tajp, ', '.join(supported_types)))
+    if tajp not in SUPPORTED_TYPES:
+        err("Type {} is not supported. The supported types are:\n "
+            "{}.".format(tajp, ', '.join(SUPPORTED_TYPES)))
+        return 1
+
+    if direction not in ["to", "from"]:
+        err("Expected the conversion string to satisfy the format to_type or "
+            "from_type, got {}".format(conversion))
+        return 1
+
+    conversion_method = getattr(numenc, conversion)
 
     if direction == "to":
-        assert isinstance(args.value, str)
-        value = bytes.fromhex(args.value)
-    elif direction == "from":
+        if not isinstance(args.value, str):
+            err("expected string value, got {}".format(type(args.value)))
+            return 1
+        try:
+            value_bts = bytes.fromhex(args.value)
+        except ValueError:
+            err("expected a hexadecimal number, got {}".format(args.value))
+            return 1
+        result_str = str(conversion_method(value_bts)) + '\n'
+
+    else:
         if "int" in tajp:
             try:
-                value = int(args.value)
-            except:
-                raise ValueError("expected an integer, got "
-                                 "{}".format(args.value))
+                value_int = int(args.value)
+            except ValueError:
+                err("expected an integer, got {}".format(args.value))
+                return 1
+            result_str = (conversion_method(value_int)).hex() + '\n'
         elif "float" in tajp:
             try:
-                value = float(args.value)
-            except:
-                raise ValueError("expected a float, got "
-                                 "{}".format(args.value))
+                value_flt = float(args.value)
+            except ValueError:
+                err("expected a float, got {}".format(args.value))
+                return 1
+            result_str = (conversion_method(value_flt)).hex() + '\n'
         else:
-            raise AssertionError("The type is neither integer nor float, "
-                                 "but {}".format(tajp))
-    else:
-        raise ValueError("Expected the conversion string to satisfy "
-                         "the format to_type or from_type, got "
-                         "{}".format(conversion_type))
-
-    conversion_method = getattr(numenc, conversion_type)
-    result = conversion_method(value)
-    if type(result) == bytes:
-        resultstr = result.hex() + '\n'
-    else:
-        resultstr = str(result) + '\n'
+            err("The type is neither integer nor float, but {}".format(tajp))
+            return 1
+
+    sys.stdout.write(result_str)
+    sys.stdout.flush()
+
+    return 0
+
+
+def err(message: str) -> None:
```

Comment:
I wouldn't use colored logging for a utility -- color codes are only applicable to certain families of terminals, don't look well in logs (_e.g._, you'd need to check that the program is run in a terminal and strip color codes otherwise), don't work in Windows _etc._

I think the canonical way be to write:

```python
print("...", file=sys.stderr)
```